### ISSUE TITLE
IsAny returns 'boolean'

### ISSUE THREAD

**Issue by rohit-smpx** [dsherret/conditional-type-checks#3]
#### Bug
```ts
let a: IsAny<2>;
```
`a` is of type `boolean` here.

##### Expected Behaviour:
`a` should be of type `false`

Typescript version: 3.0.1

##### Suggestion:
It works for `any` type correctly, `IsAny<any>` returns true, so we can use that to distinguish further. 
```ts
export type IsAny<T> = IsUnknown<T> extends true ? false
    : IsNever<T> extends true ? false
    : (T extends any ? any extends T ? true : false : false) extends true ? true : false;
```

**Comment by dsherret** [comment]
@rohit-smpx thanks for reporting this! Seems to return boolean for other types like `IsAny<string>`.

Do you want to submit a PR with your suggestion? That seems to fix the issue. If not, I'll fix this sometime soon.

**Comment by rohit-smpx** [comment]
> Do you want to submit a PR with your suggestion? That seems to fix the issue. If not, I'll fix this sometime soon.

Opened #4

**Comment by dsherret** [comment]
@rohit-smpx released in 1.0.1. Thank again for the fix! :)

**Comment by rohit-smpx** [comment]
No problems, thanks for this great little library!
